extensions/extensions-configuration-store-vault/src/main/java/com/boozallen/aissemble/configuration/dao/PropertyDao.java

```diff
@@ -0,0 +1,150 @@
+package com.boozallen.aissemble.configuration.dao;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Extensions::Configuration::Store::Vault
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultConfig;
+import com.bettercloud.vault.VaultException;
+import com.bettercloud.vault.api.mounts.MountPayload;
+import com.bettercloud.vault.api.mounts.MountType;
+import com.bettercloud.vault.response.LogicalResponse;
+import com.bettercloud.vault.response.MountResponse;
+import com.boozallen.aissemble.configuration.exception.PropertyDaoException;
+import com.boozallen.aissemble.configuration.store.Property;
+import com.boozallen.aissemble.data.encryption.VaultUtil;
+import com.boozallen.aissemble.data.encryption.config.DataEncryptionConfiguration;
+import org.aeonbits.owner.KrauseningConfigFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class PropertyDao implements Dao<Property> {
+    private static final Logger logger = LoggerFactory.getLogger(PropertyDao.class);
+
+    @Override
+    public boolean checkEmpty() {
+        throw new RuntimeException("not implemented");
+    }
+
+    /**
+     * Read the property from Vault server with given group name and property name
+     * @param groupName to read from the vault
+     * @param propertyName to read from the vault
+     */
+    @Override
+    public Property read(String groupName, String propertyName) {
+        try {
+            final Vault vault = getVault();
+            // read from Vault
+            final LogicalResponse readResponse = vault.logical()
+                    .read(String.format("properties/%s/%s", groupName, propertyName));
+
+            return new Property(groupName, propertyName, readResponse.getData().get(propertyName));
+        } catch (VaultException e) {
+            throw new PropertyDaoException(String.format("Unable to read from vault with groupName: %s and property name: %s.", groupName, propertyName), e);
+        }
+    }
+
+    /**
+     * Write given property to the vault
+     * @param property to be written to vault
+     * @throws PropertyDaoException
+     */
+    @Override
+    public void write(Property property) throws PropertyDaoException {
+        if (property != null) {
+            try {
+                final Vault vault = getVault();
+                this.write(vault, Collections.singletonList(property));
+            } catch (VaultException e) {
+                throw new PropertyDaoException(String.format("Unable to write to vault with property: %s.", property), e);
```

Comment:
C: Same note about not logging values.  The `toString` of `Property` includes the value. I think we probably should just remove the value from `toString` all together to reduce the risk of leaking information, but regardless we shouldn't rely on it here.  Just print the property group/name directly.